src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -362,25 +364,72 @@ public Info info() throws DockerException, InterruptedException {
     WebTarget resource = resource()
         .path("containers").path("json");
 
+    final Map<String, List<String>> filters = newHashMap();
     for (ListContainersParam param : params) {
-      resource = resource.queryParam(param.name(), param.value());
+      if (param instanceof ListContainersFilterParam) {
+        List<String> filterValueList;
+        if (filters.containsKey(param.name())) {
+          filterValueList = filters.get(param.name());
+        } else {
+          filterValueList = Lists.newArrayList();
+        }
+        filterValueList.add(param.value());
+        filters.put(param.name(), filterValueList);
+      } else {
+        try {
+          final String encodedName = URLEncoder.encode(param.name(), UTF_8.name());
+          final String encodedValue = URLEncoder.encode(param.value(), UTF_8.name());
+          resource = resource.queryParam(encodedName, encodedValue);
+        } catch (UnsupportedEncodingException e) {
+          throw new DockerException(e);
+        }
+      }
+    }
+
+    // If filters were specified, we must put them in a JSON object and pass them using the
+    // 'filters' query param like this: filters={"dangling":["true"]}
```

Comment:
You can see that the code I replaced was also writing out a JSON list for each key in filters. Lines 395–397:
```
generator.writeArrayFieldStart(entry.getKey());		
generator.writeString(entry.getValue());		
generator.writeEndArray();
```
But it assumed there would only ever be one value for each key, which isn't always the case.